### PR TITLE
[ubuntu] Change docker to install from official repository

### PR DESCRIPTION
**Pull request by shamil-mubarakshin**
# Description
Install `docker` from official repository

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/linux/scripts/installers/docker.sh [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-    echo "Docker ($docker_package) is already installed"
-fi
+repo_url="https://download.docker.com/linux/ubuntu"
+gpg_key="/usr/share/keyrings/docker.gpg"
+repo_path="/etc/apt/sources.list.d/docker.list"
+
+apt-get update
+apt-get install ca-certificates curl gnupg2
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o $gpg_key
+echo "deb [arch="$(dpkg --print-architecture)" signed-by=$gpg_key] $repo_url "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" > $repo_path
```
Comment (mikhailkoliada) [inline images/linux/scripts/installers/docker.sh:17]: regarding arch, not that we have different one, so we can put `amd64` explicitly

File: images/linux/scripts/installers/docker.sh [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
-else
-    echo "Docker ($docker_package) is already installed"
-fi
+repo_url="https://download.docker.com/linux/ubuntu"
+gpg_key="/usr/share/keyrings/docker.gpg"
+repo_path="/etc/apt/sources.list.d/docker.list"
+
+apt-get update
+apt-get install ca-certificates curl gnupg2
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o $gpg_key
```
Comment (mikhailkoliada) [inline images/linux/scripts/installers/docker.sh:16]: we can probably use the `os.sh` helper and use the `getOSVersionLabel` function to get a code name

File: images/linux/scripts/installers/docker.sh [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
-    apt-get install -y moby-compose
-else
-    echo "Docker ($docker_package) is already installed"
-fi
+repo_url="https://download.docker.com/linux/ubuntu"
+gpg_key="/usr/share/keyrings/docker.gpg"
+repo_path="/etc/apt/sources.list.d/docker.list"
+
+apt-get update
+apt-get install ca-certificates curl gnupg2
```
Comment (mikhailkoliada) [inline images/linux/scripts/installers/docker.sh:16]: you do not need to install those explicitly, they are already
pre-installed earlier

File: images/linux/scripts/installers/docker.sh [hunk 4]
```diff
@@ -1,25 +1,28 @@
...
-    apt-get install --no-install-recommends -y moby-buildx
-    apt-get install -y moby-compose
-else
-    echo "Docker ($docker_package) is already installed"
-fi
+repo_url="https://download.docker.com/linux/ubuntu"
+gpg_key="/usr/share/keyrings/docker.gpg"
+repo_path="/etc/apt/sources.list.d/docker.list"
+
+apt-get update
```
Comment (mikhailkoliada) [inline images/linux/scripts/installers/docker.sh:15]: apt-get is not needed here as well

**Review by mikhailkoliada — APPROVED** [review]

File: images/linux/scripts/installers/docker.sh [hunk 5]
```diff
@@ -1,25 +1,26 @@
...
+
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o $gpg_key
+echo "deb [arch=amd64 signed-by=$gpg_key] $repo_url $(getOSVersionLabel) stable" > $repo_path
+apt-get update
+apt-get install --no-install-recommends docker-ce docker-ce-cli containerd.io docker-buildx-plugin
+
+# Install docker compose v2 from releases
+URL=$(get_github_package_download_url "docker/compose" "contains(\"compose-linux-x86_64\")")
+curl -fsSL $URL -o /usr/libexec/docker/cli-plugins/docker-compose
+chmod +x /usr/libexec/docker/cli-plugins/docker-compose
```
Comment (josegonzalez) [inline images/linux/scripts/installers/docker.sh:23]: Why not install docker-compose from the apt repository? I'm asking as I'm seeing failures in the [Dokku project CI](https://github.com/dokku/dokku) after the runner-images update (we're failing to upgrade docker-compose).

Comment (shamil-mubarakshin) [reply images/linux/scripts/installers/docker.sh:23]: Docker compose packages in both `packages.microsoft.com` and `download.docker.com` tend to lag behind the version from github releases, resulting in fixes arriving late

**Pull request merged**